### DISCUSSION TITLE
bilinear form for divergence operator

### DISCUSSION THREAD

**Discussion by wangzuodong1997** [kinnala/scikit-fem#891] [Q&A] [answered]
Hello everyone, I have a question about bilinear form of Darcy problem:
I want to have a bilinearform as follows:

@BilinearForm
def a(u,v,w):
      return div(b_vector_field * grad(u)  ) * v

where b_vector_field is defined as follows:

def b_vector_field(x: np.ndarray):
    return DiscreteField(np.array([x[0],x[1]]))

After these definitions, I want to assembling the matrix but it has some errors:

asm(a,basis)
ValueError: einstein sum subscripts string contains too many subscripts for operand 1

Anyone can help me to fix this problem? Thank you very much!

**Comment by kinnala** [comment, marked as answer]
So you want to find the second derivative of u? This is possible but you need to use e.g. ElementTriArgyris (or other subclass of ElementGlobal). Moreover, scikit-fem cannot chain grad and div like that because there is no symbolic engine under the hood. This means you need to apply product rule yourself and find grad of b also by hand. If you give more details I can help better.

**Reply by wangzuodong1997** [reply]
OK, thank you very much!
